torch/csrc/api/include/torch/nn/init.h

```diff
@@ -1,27 +1,32 @@
 #pragma once
 
+#include <c10/util/variant.h>
 #include <torch/csrc/WindowsTorchApiMacro.h>
+#include <torch/enum.h>
 #include <torch/types.h>
 
 namespace torch {
 namespace nn {
 namespace init {
 
-enum class Nonlinearity {
-  Linear,
-  Conv1D,
-  Conv2D,
-  Conv3D,
-  ConvTranspose1D,
-  ConvTranspose2D,
-  ConvTranspose3D,
-  Sigmoid,
-  Tanh,
-  ReLU,
-  LeakyReLU
-};
-
-enum class FanMode { FanIn, FanOut };
+using Nonlinearity = c10::variant<
+  enumtype::torch::kLinear,
```

Comment:
@yf225 ```enumtype::torch::``` doesn't seem right. Why are you using torch namespace both outside and inside ```enumtype```? I think that the inner ```torch``` is unnecessary.

Reply:
Sounds good, I changed it to `enumtype::kLinear`.